spec/src/main/asciidoc/cdi.asciidoc

```diff
@@ -19,4 +19,158 @@
 
-TODO write the documentation after discussion on Config annotations is resolved
\ No newline at end of file
+=== Injecting `ManagedExecutor`
+
+Application can inject instances of `ManagedExecutor` using CDI with the option to share same executors between multiple injection points.
+The simplest use case will create a new executor instance for every such injection point and the configuration will be equal to default values of `@ManagedExecutorConfig` annotation.
+
+[source, java]
+----
+@Inject
+ManagedExecutor executor;
+----
+
+==== Configuring `ManagedExecutor`
+
+Oftentimes there is need to fine-grain the executor instance and change the number of maximum asynchronous tasks or define what contexts are to be propagated.
+This is where `@ManagedExecutorConfig` annotation comes into play allowing to configure all the aspects of an injected `ManagedExecutor` instance.
+Note that using just this annotation, new instance of `ManagedExecutor` is created for every injection point even should those injection points have equal configuration.
+
+[source, java]
+----
+@Inject
+@ManagedExecutorConfig(maxAsync = 5, propagated = ThreadContext.SECURITY)
+ManagedExecutor configuredExecutor;
+----
+
+==== Sharing `ManagedExecutor` Instances
+
+Last but not least, sometimes it is desirable to share the same instance of `ManagedExecutor` among several injection points.
+For such use case, the specification provides a CDI qualifier `@NamedInstance` which takes one parameter - a unique name assigned to the given executor configuration.
+Having one injection point that defines both, the name using `@NamedInstance` and the configuration using `@ManagedExecutorConfig`, application can then inject the same instance repeatedly using the qualifier.
+Assuming two beans wish to share an executor instance, the code looks as follows:
+
+[source, java]
+----
+@ApplicationScoped
+public class MyFirstBean {
+
+  @Inject
+  @NamedInstance("myExecutor")
+  @ManagedExecutorConfig(maxAsync = 5, propagated = ThreadContext.SECURITY)
+  ManagedExecutor executor;
+
+}
+
+@ApplicationScoped
+public class MySecondBean {
+
+  @Inject
+  @NamedInstance("myExecutor")
+  ManagedExecutor sharedExecutor;
+
+}
+----
+
+Since the configuration of `ManagedExecutor` is optional, a shared `ManagedExecutor` with default configuration is also a possibility.
+
+[source, java]
+----
+@ApplicationScoped
+public class FooBean {
+
+  @Inject
+  @NamedInstance("sharedDefaultME")
+  ManagedExecutor executor;
+
+}
+
+@ApplicationScoped
+public class BarBean {
+
+  @Inject
+  @NamedInstance("sharedDefaultME")
+  ManagedExecutor sharedExecutor;
+
+}
+----
+
+If multiple configurations of `ManagedExecutor` are assigned the same name, the container detects it and raises a `javax.enterprise.inject.spi.DefinitionException`.
+
+=== Injecting `ThreadContext`
+
+Application can inject instances of `ThreadContext` using CDI with the option to share same `ThreadContext` between multiple injection points.
```

Comment:
Similar to comment on Injecting Manager, this sentence is confusing where currently placed (introducing an example of not sharing).  I'd suggest moving it to the section on "Sharing `ThreadContext` Instances", but you already covered this information there, so it seems best to just omit from the current section.